### PR TITLE
MRG: Fix writing preloaded data of a non-BIDS format with automated format conversion

### LINKED ISSUE #1101: ENH: fix manufacturer checking in _channels_tsv()
### Description of the problem

in `write_bids_raw`, `allow_preload=True` should allow writing EEG files from unsupported manufacturers / files formats (e.g. EGI .raw).
Yet, _channels_tsv() makes strong assumptions about extensions e.g. when looking up ignored channels.
Could it make sense to implement `IGNOREd_CHANNELS` as a default dict or only look that up if the extension is known?

### Steps to reproduce

```Python

import tempfile
import mne
from mne_bids import write_raw_bids, BIDSPath

fname = mne.datasets.testing.data_path() / 'montage' / 'egi_dig_test.raw'

raw = mne.io.read_raw_egi(fname)

with tempfile.TemporaryDirectory() as tmpdir:
    bp = BIDSPath(subject='1', root=tmpdir, datatype='eeg', task='foo')

    write_raw_bids(raw, bp, format='BrainVision', allow_preload=True)
```

### Expected results

converting to BrainVision from EGI .raw (once preloaded)

### Actual results

```
channels_tsv(raw, fname, overwrite)
    113 # get the manufacturer from the file in the Raw object
    114 _, ext = _parse_ext(raw.filenames[0])
--> 115 manufacturer = MANUFACTURERS[ext]
    117 ignored_channels = IGNORED_CHANNELS.get(manufacturer, list())
    119 status, ch_type, description = list(), list(), list()

KeyError: '.raw'
```

### Additional information

Platform:         Linux
Python:           3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10)  [GCC 10.3.0]
mne:              1.2.0
numpy:            1.23.4 {unknown linalg bindings (threadpoolctl module not found: No module named 'threadpoolctl')}
scipy:            1.9.1

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #1101

@dengemann Can you give this a try?

PR Description
--------------

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [x] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/write.py [hunk 1]
```diff
@@ -112,8 +112,7 @@ def _channels_tsv(raw, fname, overwrite=False):
 
     # get the manufacturer from the file in the Raw object
     _, ext = _parse_ext(raw.filenames[0])
-    manufacturer = MANUFACTURERS[ext]
-
+    manufacturer = MANUFACTURERS.get(ext, '')
```
Comment (dengemann) [inline mne_bids/write.py:115]: yessss, this looks good. exactly what I had in mind.

Comment (dengemann) [reply mne_bids/write.py:115]: to test this, we could actually recycle my minimal example into a unit test. WDYT?

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
Not sure why that one CI job failed, I just restarted it. Let's see…

**Comment by hoechenberger** [comment]
Found the problem. Should pass now!!

**Comment by hoechenberger** [comment]
@agramfort @sappelhoff Feel free to merge once green, or enable automerge :)

**Review by adam2392 — APPROVED** [review]
Nice fix!

**Comment by agramfort** [comment]
thx @hoechenberger

**Comment by dengemann** [comment]
Thanks everyone!!

**Pull request merged**